tests/test_linear_model.py

```diff
@@ -172,6 +172,42 @@ def test_ridge_regression_2fold_cv_small_to_large(
             f"error {err} surpasses threshold for zero {self.eps}",
         )
 
+    @parameterized.expand(ridge_parameters)
+    def test_ridge_regression_2fold_regularization(
+        self, name, alpha_type, regularization_method
+    ):
+        # tests if the regularization in the CV split of
+        # RidgeRegression2FoldCV does effect the results
+
+        # regularization parameters are chosen to match the singular values o
+        # the features, thus each regularization parameter affects the minimized
+        # weight matrix and thus the error
+        _, singular_values, _ = np.linalg.svd(self.features_all)
+        if alpha_type == "absolute":
+            alphas = singular_values[1:][::-1]
+        if alpha_type == "relative":
+            alphas = singular_values[1:][::-1] / singular_values[0]
+
+        # tests if RidgeRegression2FoldCV does do regularization correct
+        ridge = RidgeRegression2FoldCV(
+            alphas=alphas,
+            alpha_type=alpha_type,
+            regularization_method=regularization_method,
+            scoring="neg_root_mean_squared_error",
+        ).fit(self.features_all, self.features_all)
+        twofold_rmse = -np.array(ridge.cv_values_)
+
+        # since the data can be perfectly reconstructed,
+        # larger regularization parameters (alphas) should result in
+        # larger errors
+        error_grad = twofold_rmse[1:] - twofold_rmse[:-1]
+        self.assertTrue(
+            np.all(error_grad > 0),
```

Comment:
Should this be some thresholded value, say 1E-8?

Reply:
added threshold